dissect/target/plugins/os/windows/_os.py

```diff
@@ -110,25 +112,137 @@ def ips(self) -> list[str]:
+            reg_key = self.target.registry.key(self.CURRENT_VERSION_KEY)
+            version = reg_key.value("CurrentMajorVersionNumber").value
         except RegistryError:
             pass
 
+        return version
+
+    @internal
+    def minor_version(self) -> Optional[int]:
+        """Return the NT minor version number (starting from NT 10.0 / Windows 10).
+
+        Returns:
+            The integer value of the NT minor version number or ``None`` if the
+            CurrentMinorVersionNumber sub key is not present or any other
+            registry error occurred.
+        """
+        version = None
         try:
-            r = self.target.registry.key(key)
-            product_name = r.value("ProductName").value
-            current_version = r.value("CurrentVersion").value
-            current_build_number = r.value("CurrentBuildNumber").value
-            return f"{product_name} (NT {current_version}) {current_build_number} {csd_version}"
+            reg_key = self.target.registry.key(self.CURRENT_VERSION_KEY)
+            version = reg_key.value("CurrentMinorVersionNumber").value
         except RegistryError:
             pass
 
+        return version
+
+    @internal
+    def nt_version(self) -> Optional[int]:
+        """Return the Windows NT version in x.y format.
+
+        For systems up to and including NT 6.3 (Win 8 / Win 2012 Server) this
+        will be the value of the CurrentVersion sub key.
+
+        For systems starting from NT 10.0 (Win 10) this will be a value
+        constructed from the combination of the CurrentMajorVersionNumber and
+        CurrentMinorVersionNumber sub keys.
+
+        Returns:
+            The string value of the NT version or ``None`` if the any of the
+            sub keys are not present or any other registry error occurred.
+        """
+        version = None
+
+        major_version = self.major_version()
+        if major_version is not None:
+            minor_version = self.minor_version()
+            if minor_version is None:
+                minor_version = ""
+            version = f"{major_version}.{minor_version}"
+        else:
+            version = self._legacy_current_version()
+
+        return version
+
+    @export(property=True)
+    def version(self) -> Optional[str]:
+        # Note that We don't show the "version" (aka FeatureRelease) as shown
+        # by WinVer.exe, whic uses the keys:
+        # ReleaseId: for Win 10 ReleaseId <= 2004
+        # DisplayVersion: for Win 10 ReleaseId >= 2009 (DisplayVersion = 20H2
+        #                 in this case)
+        #
+        # https://www.vcloudinfo.com/2020/12/how-to-decode-windows-version-numbers.html
+        version_string = None
+
+        try:
+            reg_key = self.target.registry.key(self.CURRENT_VERSION_KEY)
+
+            # Service Pack Version, when installed using the normal method (may not be present).
+            try:
+                csd_version = reg_key.value("CSDVersion").value
```

Comment:
Side-note, but maybe it's worth adding a `default` option to the `value()` function at some point, just like dictionaries.

Reply:
That would help with a lot of boilerplate yes.

Reply:
Could you create a ticket for that as well?